pkgerrors.go

```diff
@@ -8,50 +8,112 @@ import (
 
 type pkgError struct {
 	Err        error
-	Message    string
+	Messages   []string
 	StackTrace StackTrace
 }
 
 const (
 	pkgErrorsMessageDelimiter = ": "
 )
 
+// convertPkgError converts pkg/errors to fail.
+// It returns nil if the error is not derived from pkg/errors.
+func convertPkgError(err error) (convertedErr *Error) {
+	pkgErr := extractPkgError(err)
+	if pkgErr == nil {
+		return
+	}
+
+	if appErr, ok := pkgErr.Err.(*Error); ok {
+		convertedErr = appErr.Copy()
+		convertedErr.StackTrace = mergeStackTraces(appErr.StackTrace, pkgErr.StackTrace)
```

Comment:
In the original logic, all of the annotations done by `pkg/errors` were thrown away.
Now it's handled properly regardless of the value of `pkgErr.Err.(*Error)`.

https://github.com/srvc/fail/blob/b23d9d32fb81b39e8baaa9517ed882d9abf03927/error.go#L100-L115

```go
	if appErr, ok := pkgErr.Err.(*Error); ok {
		wrappedErr = appErr.Copy()
		// <--- Discarding messages and stack trace annotated with pkg/errors.
	} else {
```